ui/modModules/multiplayer/multiplayer.js

```diff
@@ -693,16 +695,84 @@ function($scope, $state, $timeout, $mdDialog, $filter, ConfirmationDialog, toast
 
 	};
 
+	vm.downloadingMods = [];
+	var lastModInfo = ''
+
 	$scope.$on('LoadingInfo', function (event, data) {
-		if (document.getElementById('LoadingStatus').innerText != data.message) console.log(data.message)
-		if (data.message == "done") {
-			document.getElementById('LoadingStatus').innerText = "Done";
+		//console.log(data.message)
+
+		// Split the message into parts: mod number, mod name, progress, speed
+		let modNumber = null;
+		let modName = null;
+		let progress = null;
+		let speed = null;
+
+		if (data.message.startsWith("Downloading Resource")) {
+			// Sample: 'Downloading Resource 1/10: Nissan 350z.zip (1.0%) at 12.8 Mbit/s'
+			// Extract mod number, name, progress, and speed from the message
+			const regex = /Downloading Resource (\d+\/\d+): (.+?) \((\d+\.\d+)%\)(?: at (.+))?/;
+			const matches = data.message.match(regex);
+			if (matches) {
+				modNumber = matches[1];
+				modName = matches[2];
+				progress = matches[3];
+				speed = matches[4] || '...';
+			}
+			//console.log(`Mod ${modNumber}: ${modName} - ${progress}% at ${speed}`);
+
+			// Update current downloading mod info and if complete then push this mod into the downloaded mods info
+			$scope.$apply(function() {
+				// Update or add the current mod being downloaded
+				let existingMod = vm.downloadingMods.find(mod => mod.name === modName);
+				if (existingMod) {
+					existingMod.progress = progress;
+					existingMod.speed = speed;
+				} else {
+					// add this new mod to the beginning of the array
+					vm.downloadingMods = [{ number: modNumber, name: modName, progress: progress, speed: speed }, ...vm.downloadingMods];
+				}
+
+				// If we switched to a new mod, mark the last one as done
+				if (lastModInfo != '' && lastModInfo != modName) {
+					let lastMod = vm.downloadingMods.find(mod => mod.name === lastModInfo);
+					lastMod.progress = 100;
+					lastMod.speed = 'Done';
+					lastModInfo = modName;
+				}
+			});
+		} else if (data.message.startsWith("Loading Resource")) {
+			// Sample: 'Loading Resource 1/70: Scintillacamaf.zip'
+			const regex = /Loading Resource (\d+\/\d+): (.+)/;
+			const matches = data.message.match(regex);
+			if (matches) {
+				modNumber = matches[1];
+				modName = matches[2];
+			}
+			//console.log(`Mod ${modNumber}: ${modName} - Loading`);
+
+			// Update current downloading mod info and if complete then push this mod into the downloaded mods info
+			$scope.$apply(function() {
+				// Update or add the current mod being downloaded
+				let existingMod = vm.downloadingMods.find(mod => mod.name === modName);
+				if (existingMod) {
+					existingMod.progress = '100';
+					existingMod.speed = 'Downloaded';
+				} else {
+					vm.downloadingMods = [{ number: modNumber, name: modName, progress: '100', speed: 'Loading' }, ...vm.downloadingMods];
```

Comment:
Same thing as above, maybe unshift here? Also Maybe the speed should be `Downloaded` here(?)


```suggestion
					vm.downloadingMods.unshift({ number: modNumber, name: modName, progress: '100', speed: 'Downloaded' });
```

Reply:
I like the idea of Downloaded but there is a lacking in state for each mod from the launcher currently.
If we could address that I would love to make it Downloaded and then loaded / ready when it is loaded.

Will update it to Downloaded though in the meantime.